src/main/kotlin/com/cognifide/gradle/aem/deploy/tasks/AbstractTask.kt

```diff
@@ -169,4 +176,79 @@ abstract class AbstractTask : DefaultTask(), AemTask {
         return response
     }
 
+    protected fun deletePackage(installedPackagePath: String, sync: DeploySynchronizer) {
+        val url = sync.htmlTargetUrl + installedPackagePath + "/?cmd=delete"
+
+        logger.info("Deleting package using command: " + url)
+
+        try {
+            val rawHtml = sync.post(url)
+            val response = DeleteResponse(rawHtml)
+
+            when (response.status) {
+                AbstractHtmlResponse.Status.SUCCESS,
+                AbstractHtmlResponse.Status.SUCCESS_WITH_ERRORS -> if (response.errors.isEmpty()) {
+                    logger.info("Package successfully deleted.")
+                } else {
+                    logger.warn("Package deleted with errors.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package deleted with errors!")
+                }
+                AbstractHtmlResponse.Status.FAIL -> {
+                    logger.error("Package deleting failed.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package deleting failed!")
+                }
+            }
+
+        } catch (e: Exception) {
+            throw DeployException("Cannot delete package.", e)
+        }
+    }
+
+    protected fun uninstallPackage(installedPackagePath: String, sync: DeploySynchronizer) {
+        val url = sync.htmlTargetUrl + installedPackagePath + "/?cmd=uninstall"
+
+        logger.info("Uninstalling package using command: " + url)
+
+        try {
+            val rawHtml = sync.post(url, mapOf(
+                    "recursive" to config.recursiveInstall,
+                    "acHandling" to config.acHandling
+            ))
+            val response = UninstallResponse(rawHtml)
+
+            when (response.status) {
+                AbstractHtmlResponse.Status.SUCCESS,
+                AbstractHtmlResponse.Status.SUCCESS_WITH_ERRORS -> if (response.errors.isEmpty()) {
+                    logger.info("Package successfully uninstalled.")
+                } else {
+                    logger.warn("Package uninstalled with errors.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package uninstalled with errors!")
+                }
+                AbstractHtmlResponse.Status.FAIL -> {
+                    logger.error("Package uninstalling failed.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package uninstalling failed!")
+                }
+            }
+
+        } catch (e: Exception) {
+            throw DeployException("Cannot uninstall package.", e)
+        }
+    }
+
+    protected fun ensureUserAwareness(taskName: String) {
+        if (project.properties.containsKey(USER_AWARENESS_PROP)) return
```

Comment:
```
val force = project.properties[FORCE_PROP] as String?

if (force == null || !BooleanUtils.isFalse(force)) {
   throw DeployException()
}
```

Imagine that you are generating bash script. and in some edge case you want to specify `aem.force=false` but it will not work in your case. Imagine that another lazy developer will copy your lines for another feature in which false case will be possible. Do it right from the start! :>